crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -0,0 +1,1150 @@
+//! Ruleset utility struct
+
+use ruma::{
+    api::client::push::RuleScope,
+    push::{
+        Action, NewConditionalPushRule, NewPushRule, NewSimplePushRule, PredefinedContentRuleId,
+        PredefinedOverrideRuleId, PredefinedUnderrideRuleId, PushCondition, RemovePushRuleError,
+        RuleKind, Ruleset, Tweak,
+    },
+    RoomId,
+};
+
+use super::RoomNotificationMode;
+use crate::error::NotificationSettingsError;
+
+/// enum describing the commands required to modify the owner's account data.
+#[derive(Clone, Debug)]
+pub(crate) enum Command {
+    /// Set a new push rule
+    SetPushRule(RuleScope, NewPushRule),
+    /// Set whether a push rule is enabled
+    SetPushRuleEnabled(RuleScope, RuleKind, String, bool),
+    /// Delete a push rule
+    DeletePushRule(RuleScope, RuleKind, String),
+}
+
+pub(crate) struct Rules {
+    pub ruleset: Ruleset,
+}
+
+#[allow(dead_code)]
+impl Rules {
+    pub(crate) fn new(ruleset: Ruleset) -> Self {
+        Rules { ruleset }
+    }
+
+    /// Gets all user defined rules matching a given `room_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `room_id` - A `RoomId`
```

Comment:
This adds nothing, let's remove it.

Reply:
All done. I've cleaned up the documentation.